ProcessLib/UniformDirichletBoundaryCondition.h

```diff
@@ -67,16 +69,13 @@ class UniformDirichletBoundaryCondition
                                 MeshLib::MeshItemType::Node,
                                 id);
             // TODO: that might be slow, but only done once
-            id = dof_table.getGlobalIndex(l, component_id);
+            const auto g_idx = dof_table.getGlobalIndex(l, component_id);
+            if (g_idx >= 0)
+            {
+                global_ids.emplace_back(g_idx);
```

Comment:
Possibly it would be slightly more performant to reserve space for all entries of `ids`. Sure, that would waste some space, but memory consumption won't blow up by that.
But I have not measured the performace impact and this function is only called once per simulation (or in future maby once per timestep).
So in my opinion: It would be nice if you kept the `global_ids.reserve(global_ids.size() + ids.size());`, at least if you do not strongly object against the waste of space.